### PR TITLE
[PW-1580] Add /payout service

### PR DESCRIPTION
**Pull request by AlexandrosMor**
Endpoints added
/storeDetailAndSubmitThirdParty
/storeDetail
/submitThirdParty
/confirmThirdParty
/declineThirdParty
/payout

Unit tests
StoreDetailAndSubmitThirdParty
StoreDetail
ConfirmThirdParty
SubmitThirdParty
DeclineThirdParty
Payout

### CODE REVIEW

File: Adyen.Test/BaseTest.cs [hunk 1]
```diff
@@ -295,7 +295,29 @@ protected Client CreateMockTestClientRequest(string fileName)
...
+            };
+            return clientMock;
+        }
+
+        /// <summary>
+        /// Creates mock test client 
+        /// </summary>
+        /// <param name="fileName">The file that is returned</param>
+        /// <returns>IClient implementation</returns>
+        protected Client CreateMockTestClientPayoutRequest(string fileName)
```
Comment (martinsrenato) [inline Adyen.Test/BaseTest.cs]: Is this method `CreateMockTestClientPayoutRequest` being used? 
It's the same as `CreateMockTestClientNullRequiredFieldsRequest` and the PayoutTest is using this second one.

File: Adyen/Service/Payout.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        {
+            var jsonRequest = Util.JsonOperation.SerializeRequest(request);
+            var jsonResponse = _storeDetailAndSubmitThirdParty.Request(jsonRequest);
+            return JsonConvert.DeserializeObject<StoreDetailAndSubmitResponse>(jsonResponse);
+        }
+
+        public ConfirmThirdPartyResponse ConfirmThirdParty(ConfirmThirdPartyRequest request)
+        {
+            var jsonRequest = Util.JsonOperation.SerializeRequest(request);
+            var jsonResponse = _storeDetailAndSubmitThirdParty.Request(jsonRequest);
```
Comment (Aleffio) [inline Adyen/Service/Payout.cs]:
```suggestion
            var jsonResponse = _confirmThirdParty.Request(jsonRequest);
```

File: Adyen/Service/Payout.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            return JsonConvert.DeserializeObject<ConfirmThirdPartyResponse>(jsonResponse);
+        }
+        
+        public DeclineThirdPartyResponse DeclineThirdParty(DeclineThirdPartyRequest request)
+        {
+            var jsonRequest = Util.JsonOperation.SerializeRequest(request);
+            var jsonResponse = _storeDetailAndSubmitThirdParty.Request(jsonRequest);
```
Comment (Aleffio) [inline Adyen/Service/Payout.cs]:
```suggestion
            var jsonResponse = _declineThirdParty.Request(jsonRequest);
```

File: Adyen/Service/Payout.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            return JsonConvert.DeserializeObject<DeclineThirdPartyResponse>(jsonResponse);
+        }
+
+        public StoreDetailResponse StoreDetail(StoreDetailRequest request)
+        {
+            var jsonRequest = Util.JsonOperation.SerializeRequest(request);
+            var jsonResponse = _storeDetailAndSubmitThirdParty.Request(jsonRequest);
```
Comment (Aleffio) [inline Adyen/Service/Payout.cs]:
```suggestion
            var jsonResponse = _storeDetail.Request(jsonRequest);
```

File: Adyen/Service/Payout.cs [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        private PayoutService _payoutService;
+
+        public Payout(Client client)
+            : base(client)
+        {
+            _storeDetailAndSubmitThirdParty = new StoreDetailAndSubmitThirdParty(this);
+            _confirmThirdParty = new ConfirmThirdParty(this);
+            _declineThirdParty = new DeclineThirdParty(this);
+            _storeDetail = new StoreDetail(this);
+            _submitThirdparty = new SubmitThirdParty(this);
```
Comment (Aleffio) [inline Adyen/Service/Payout.cs]:
```suggestion
            _submitThirdParty = new SubmitThirdParty(this);
```

File: Adyen/Service/Payout.cs [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        {
+            var jsonRequest = Util.JsonOperation.SerializeRequest(request);
+            var jsonResponse = _storeDetailAndSubmitThirdParty.Request(jsonRequest);
+            return JsonConvert.DeserializeObject<StoreDetailResponse>(jsonResponse);
+        }
+
+        public SubmitResponse SubmitThirdparty(SubmitRequest request)
+        {
+            var jsonRequest = Util.JsonOperation.SerializeRequest(request);
+            var jsonResponse = _storeDetailAndSubmitThirdParty.Request(jsonRequest);
```
Comment (Aleffio) [inline Adyen/Service/Payout.cs]:
```suggestion
            var jsonResponse = _submitThirdParty.Request(jsonRequest);
```

File: Adyen/Service/Payout.cs [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        }
+
+        public StoreDetailResponse StoreDetail(StoreDetailRequest request)
+        {
+            var jsonRequest = Util.JsonOperation.SerializeRequest(request);
+            var jsonResponse = _storeDetailAndSubmitThirdParty.Request(jsonRequest);
+            return JsonConvert.DeserializeObject<StoreDetailResponse>(jsonResponse);
+        }
+
+        public SubmitResponse SubmitThirdparty(SubmitRequest request)
```
Comment (Aleffio) [inline Adyen/Service/Payout.cs]:
```suggestion
        public SubmitResponse SubmitThirdParty(SubmitRequest request)
```

File: Adyen.Test/PayoutTest.cs [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+            Assert.AreEqual("8815131762537886", result.PspReference);
+        }
+
+        [TestMethod]
+        public void SubmitThirdPartySuccessTest()
+        {
+            Client client = CreateMockTestClientNullRequiredFieldsRequest("Mocks/payout/submitResponse-success.json");
+            Payout payout = new Payout(client);
+            SubmitRequest request = new SubmitRequest();
+            SubmitResponse result = payout.SubmitThirdparty(request);
```
Comment (Aleffio) [inline Adyen.Test/PayoutTest.cs]:
```suggestion
            SubmitResponse result = payout.SubmitThirdParty(request);
```

File: Adyen/Service/Payout.cs [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+namespace Adyen.Service
+{
+    public class Payout : AbstractService
+    {
+        private StoreDetailAndSubmitThirdParty _storeDetailAndSubmitThirdParty;
+        private ConfirmThirdParty _confirmThirdParty;
+        private DeclineThirdParty _declineThirdParty;
+        private StoreDetail _storeDetail;
+        private SubmitThirdParty _submitThirdparty;
```
Comment (Aleffio) [inline Adyen/Service/Payout.cs]:
```suggestion
        private SubmitThirdParty _submitThirdParty;
```

File: Adyen/Service/Payout.cs [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+        {
+            var jsonRequest = Util.JsonOperation.SerializeRequest(request);
+            var jsonResponse = _storeDetailAndSubmitThirdParty.Request(jsonRequest);
+            return JsonConvert.DeserializeObject<SubmitResponse>(jsonResponse);
+        }
+
+        public PayoutResponse payout(PayoutRequest request)
+        {
+            var jsonRequest = Util.JsonOperation.SerializeRequest(request);
+            var jsonResponse = _storeDetailAndSubmitThirdParty.Request(jsonRequest);
```
Comment (Aleffio) [inline Adyen/Service/Payout.cs]:
```suggestion
            var jsonResponse = _payoutService.Request(jsonRequest);
```

**Pull request merged**